### PR TITLE
Sign compare warning fixes batch 1 1

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac 

Pardon the large batch size, I am batching based on warning id cardinality, rather than the number of files.
I will batch based on number of files committed going forward.

### CODE REVIEW

File: tensorflow/compiler/xla/window_util.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@ Window MakeWindow(absl::Span<const int64> sizes,
                   absl::Span<const int64> strides) {
   Window window;
   CHECK_EQ(sizes.size(), strides.size());
-  for (auto nb = 0; nb < sizes.size(); ++nb) {
+  for (auto nb = 0; static_cast<size_t>(nb) < sizes.size(); ++nb) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/window_util.cc:45]:
```suggestion
  for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
```

File: tensorflow/core/kernels/quantization_utils.h [hunk 2]
```diff
@@ -268,7 +268,7 @@ inline void RequantizeManyInNewRangeReference(const qint32* input, int64 count,
   // that could be easily adapted for a SIMD implementation. It should also be
   // possible to perform all the calculations in 32-bit rather than 64, but
   // that's not been implemented yet.
-  for (size_t index = 0; index < count; ++index) {
+  for (size_t index = 0; static_cast<tensorflow::int64>(index) < count; ++index) {
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/quantization_utils.h:271]: Same here

File: tensorflow/core/profiler/utils/derived_timeline.cc [hunk 3]
```diff
@@ -130,7 +130,7 @@ void DerivedXLineBuilder::ExpandOrAddLevelEvent(const XEvent& event,
 }
 
 void DerivedXLineBuilder::ResetLastEvents(int level) {
-  for (int i = level; i < last_event_by_level_.size(); ++i) {
+  for (int i = level; i < static_cast<int>(last_event_by_level_.size()); ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/derived_timeline.cc:133]: And here

File: tensorflow/lite/toco/graph_transformations/convert_trivial_transpose_to_reshape.cc [hunk 4]
```diff
@@ -31,7 +31,7 @@ bool TransposeAffectsMemoryOrder(std::vector<int> perm,
   // just the shape) then the flat buffer representation shouldn't change.
   std::vector<int> old_major_index_ordering;
   std::vector<int> new_major_index_ordering;
-  for (int i = 0; i < in_shape.size(); i++) {
+  for (int i = 0; static_cast<size_t>(i) < in_shape.size(); i++) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/convert_trivial_transpose_to_reshape.cc:34]: And here

File: tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc [hunk 5]
```diff
@@ -1856,8 +1856,8 @@ void ProcessArgMinMaxOperator(Model* model, Op* op) {
   std::vector<int> output_dims;
 
   output_dims.reserve(input_dims.size() - 1);
-  for (int i = 0; i < input_dims.size(); ++i) {
-    if (i != axis) {
+  for (size_t i = 0; i < input_dims.size(); ++i) {
+    if ( static_cast<int>(i) != axis) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:1860]: And here too. It looks weird to change type of loop counter and then do the static cast immediately below.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/quantization_utils.h (hunk 2 above)
Comment (tg-at-google) [reply tensorflow/core/kernels/quantization_utils.h:271]: I'll apply the new styling to all of the changes affecting for loops.

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/compiler/xla/window_util.cc [hunk 6]
```diff
@@ -42,7 +42,7 @@ Window MakeWindow(absl::Span<const int64> sizes,
                   absl::Span<const int64> strides) {
   Window window;
   CHECK_EQ(sizes.size(), strides.size());
-  for (auto nb = 0; nb < sizes.size(); ++nb) {
+  for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/window_util.cc:45]:
```
tensorflow/compiler/xla/window_util.cc:45:21: error: expected unqualified-id before 'auto'
   for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
                     ^~~~
tensorflow/compiler/xla/window_util.cc:45:21: error: expected ';' before 'auto'
tensorflow/compiler/xla/window_util.cc:45:56: error: expected ')' before ';' token
   for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
                                                        ^
tensorflow/compiler/xla/window_util.cc:45:60: error: 'nb' was not declared in this scope
   for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
```

Let's give types here instead of `auto`

File: tensorflow/core/kernels/quantization_utils.h (hunk 2 above)
Comment (tg-at-google) [reply tensorflow/core/kernels/quantization_utils.h:271]: Actually, can you clarify @mihaimaruseac ? I am not sure of your request.

File: tensorflow/compiler/xla/window_util.cc [hunk 7]
```diff
@@ -42,7 +42,7 @@ Window MakeWindow(absl::Span<const int64> sizes,
                   absl::Span<const int64> strides) {
   Window window;
   CHECK_EQ(sizes.size(), strides.size());
-  for (auto nb = 0, auto size = sizes.size(); nb < size; ++nb) {
+  for (int nb = 0, auto size = sizes.size(); nb < size; ++nb) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/window_util.cc:45]:
```suggestion
  for (int nb = 0, int size = sizes.size(); nb < size; ++nb) {
```

File: tensorflow/core/kernels/quantization_utils.h (hunk 2 above)
Comment (mihaimaruseac) [reply tensorflow/core/kernels/quantization_utils.h:271]: Was suggesting to replace all `for (type var = 0; static_cast<other_type>(var) < limit_expr; ...)` to `for (type var = 0, type limit = limit_expr; var < limit; ...)` if `limit_expr` is an expression. Otherwise, if `limit_expr` is just a variable, we can change type of `var` or `limit_expr` variable at declaration so that they match.

**Comment by tg-at-google** [comment]
@joker-eph There are no mlir related changes in this PR.

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc [hunk 8]
```diff
@@ -1856,7 +1856,8 @@ void ProcessArgMinMaxOperator(Model* model, Op* op) {
   std::vector<int> output_dims;
 
   output_dims.reserve(input_dims.size() - 1);
-  for (int i = 0; i < input_dims.size(); ++i) {
+
+  for (int i = 0, iter_limit = input_dims.size(); i < iter_limitd; ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:1860]:
```
tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:1860:55: error: 'iter_limitd' was not declared in this scope
   for (int i = 0, iter_limit = input_dims.size(); i < iter_limitd; ++i) {
                                                       ^~~~~~~~~~~
tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:1860:55: note: suggested alternative: 'iter_limit'
   for (int i = 0, iter_limit = input_dims.size(); i < iter_limitd; ++i) {
                                                       ^~~~~~~~~~~
                                                       iter_limit
```

```suggestion
  for (int i = 0, iter_limit = input_dims.size(); i < iter_limit; ++i) {
```

**Comment by tg-at-google** [comment]
This is pesky. There were 35 files, and now only the mlir files are showing.

File: tensorflow/compiler/mlir/lite/quantization/import_quant_stats_pass.cc [hunk 9]
```diff
@@ -76,8 +76,9 @@ class ImportQuantStatsPass
   // If the index is out of range, this method returns false. Otherwise it
   // returns true if the value is a float tensor.
   bool IsQuantizableResult(Operation *op, int index) {
-    if (index < 0 || index >= static_cast<int>(op->getNumResults()))
+    if (index < 0 || index >= static_cast<int>(op->getNumResults())) {
       return false;
+    }
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/lite/quantization/import_quant_stats_pass.cc:81]: This seems unrelated?

File: tensorflow/compiler/mlir/xla/ir/hlo_ops.cc [hunk 10]
```diff
@@ -1122,6 +1122,7 @@ static LogicalResult Verify(MapOp op) {
   // increasing.
   auto values = op.dimensions().getValues<int64_t>();
   auto dimensions = std::vector<int64_t>{values.begin(), values.end()};
+
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/ir/hlo_ops.cc:1125]: remove

File: tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc [hunk 11]
```diff
@@ -1856,7 +1856,8 @@ void ProcessArgMinMaxOperator(Model* model, Op* op) {
   std::vector<int> output_dims;
 
   output_dims.reserve(input_dims.size() - 1);
-  for (int i = 0; i < input_dims.size(); ++i) {
+
```
Comment (joker-eph) [inline tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:1859]: remote the empty line

File: tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc [hunk 12]
```diff
@@ -2010,8 +2011,8 @@ void ProcessUnpackOperator(Model* model, UnpackOperator* op) {
   std::vector<int> output_dims;
 
   output_dims.reserve(input_dims.size() - 1);
-  for (int i = 0; i < input_dims.size(); ++i) {
-    if (i != op->axis) {
+  for (int i = 0, iter_limit = input_dims.size(); i < iter_limit; ++i) {
+    if ( i != op->axis) {
```
Comment (joker-eph) [inline tensorflow/lite/toco/graph_transformations/propagate_fixed_sizes.cc:2015]: remove the empty space
